### PR TITLE
Graceful shutdown

### PR DESCRIPTION
**Pull request by cyBerta**
some improvements wrt to shutting down yggmail gracefully

### CODE REVIEW

File: mobile/yggmail/yggmail.go [hunk 1]
```diff
@@ -23,67 +24,106 @@ import (
...
 
-func (ym *Yggmail) openDatabase() *error {
-	storage, err := sqlite3.NewSQLite3StorageStorage(ym.DatabaseName)
-	if err != nil {
-		return &err
+func (ym *Yggmail) setState(stateConst int) {
+	if stateConst == Running || stateConst == ShuttingDown || stateConst == Stopped || stateConst == Error {
+		ym.locker.Lock()
+		ym.state = stateConst
+		ym.locker.Unlock()
```
Comment (GRbit) [inline mobile/yggmail/yggmail.go:69]: I think you can use `defer` here as in the `GetState()` function. There will be almost no difference, but the code will look consistent.

File: mobile/yggmail/yggmail.go [hunk 2]
```diff
@@ -234,13 +288,26 @@ func (ym *Yggmail) Stop() {
...
+		ym.transport.Stop()
+		ym.transport = nil
+	}
+	ym.closeDatabase()
+
+	ym.setState(Stopped)
 }
 
-func (ym *Yggmail) sendError(errorId int, format string, a ...interface{}) {
+func (ym *Yggmail) handleError(errorId int, format string, a ...interface{}) {
```
Comment (GRbit) [inline mobile/yggmail/yggmail.go:300]: As a suggestion for errors wrapping:

You can change this to 
```Go
func (ym *Yggmail) handleError(errorId int, err error) {
```
And call this as:
```Go
ym.handleError(ErrCode, xerrors.Errorf("Error description with some useful info: %w", err))
```
And as a result, you will get a detailed error object with an optional stack trace.

To print a stacktrace with all information inside your error you can use `"%v+"` placeholder, like this:
```Go
ym.Logger.LogError(errorId, fmt.Sprintf("[  Yggmail  ] %+v", err))
```

Comment (cyBerta) [reply mobile/yggmail/yggmail.go:300]: sounds good to me. This is probably a change that should be done in the overall project, for consistency and better debugging capabilities.

Comment (cyBerta) [reply mobile/yggmail/yggmail.go:300]: Let's discuss that change on the upstream repository.

**Pull request merged**